newsroom/search.py

```diff
@@ -507,7 +507,11 @@ def prefill_search_highlights(self, search, req):
                     "lenient": True,
                 },
             )
-            elastic_highlight_query["fields"] = {"body_html": elastic_highlight_query["fields"]["body_html"]}
+            elastic_highlight_query["fields"] = {
+                "body_html": elastic_highlight_query["fields"]["body_html"],
```

Comment:
lets avoid this field specific query so it might also work with advanced search, so the fields should be like:
```py
elastic_highlight_query["fields"] = {
    "body_html": {},
    "headline": {},
    "slugline": {},
}
```
it might also need the change in the condition `if app.data.elastic.should_highlight(req) and query_string`
because we want this to work also with advanced search must contain and should contain fields

Reply:
> it might also need the change in the condition if app.data.elastic.should_highlight(req) and query_string
no, I checked we don;t need to change the condition , as it is worked with advanced search filter.